code/game/machinery/Sleeper.dm

```diff
@@ -145,12 +162,56 @@
 
 /obj/machinery/sleeper/New()
 	..()
+	RefreshParts()
 	spawn( 5 )
+		update_icon()
 		if(orient == "RIGHT")
-			icon_state = "sleeper_0-r"
+			generate_console(get_step(get_turf(src), WEST))
+		else
+			generate_console(get_step(get_turf(src), EAST))
 		return
 	return
 
+/obj/machinery/sleeper/Destroy()
+	..()
+	qdel(connected)

```

Comment:
If you want these to not hard delete and quick delete normally, both of these need to any reference to one another removed.